src/openrct2/windows/NewRide.cpp

```diff
@@ -281,12 +279,12 @@ void window_new_ride_init_vars()
         _windowNewRideCurrentTab = WINDOW_NEW_RIDE_PAGE_TRANSPORT;
     }
 
-    for (sint16 i = 0; i < 6; i++) {
+    for (auto &i : _windowNewRideHighlightedItem) {
```

Comment:
Personally I'm not a fan of these reference loops. The original code was more obvious (though it could've used a `CountOf`)